code-challenges.test.js

```diff
@@ -26,42 +26,162 @@ const numbersArray2 = [24, 27, 30, 33, 36]
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // Pseudo code:
+    // Input: an array of words
+    // Output: an array with all the words capitalized
+    // Process: create a function using the .map higher order function and .toUpperCase method to output all the given words capitalized
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("upperCaser", () => {
+    it("takes in an array of words and returns an array with all the words capitalized", () => {
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+        expect(upperCaser(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        expect(upperCaser(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+
+// test output: threeDivider
+    // ✓  takes a object as an argument and decides if the number inside it is evenly divisible by three or not (2 ms)
+    // upperCaser
+    //   ✕ takes in an array of words and returns an array with all the words capitalized
+  
+    // ● upperCaser › takes in an array of words and returns an array with all the words capitalized
+  
+    //   ReferenceError: upperCaser is not defined
 
 // b) Create the function that makes the test pass.
 
+
+// Green: 
+// const upperCaser = (array) => {
+//     let newArray = []
+//     for (let i = 0; i < array.length; i++) {
+//         newArray.push(array[i].charAt(0).toUpperCase() + array[i].slice(1))
+//     }
+//     return newArray
+// }
+
+// Refactor:
+const upperCaser = (array) => {
+    return array.map(value => value[0].toUpperCase() + value.substring(1))
+}
+
+// Notes: My original function was actually the .map refactor. It was very intuitive. I had to search for help, for the 'for loop' version of the function. The built in methods got confusing compared to the .map function. 
+
+// // output: PASS  ./code-challenges.test.js
+//   threeDivider
+//   ✓  takes a object as an argument and decides if the number inside it is evenly divisible by three or not (3 ms)
+// upperCaser
+//   ✓ takes in an array of words and returns an array with all the words capitalized (1 ms)
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
-// Pseudo code:
+// Pseudo code: 
+    // Input: a string
+    // Output: the index number of the first vowel in the input string
+    // Process: create a function that uses a for loop to iterate through a string and checks whether the string contains any vowels from the variable vowels = ["a", "e", "i", "o", "u"]
```

Comment:
Very close! You're not checking if the string contains a vowel from the vowel array, but you're actually checking the vowel array to see if it contains each character in the string through iteration with a for loop. I would also include in your pseudo code how you are going to do what you said you are going to do: you did not say you would use `.includes` on the array.